### PR TITLE
[Feature] Support create mv from mv

### LINKED ISSUE #11446: Support for creating materialized views from materialized views
## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by blackstar-baba**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11446 

## Problem Summary(Required) ：

Support create mv from mv

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 6 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|1|5|20.00%|[497, 498, 499, 500]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|1|1|100.00%|[]|

**Comment by blackstar-baba** [comment]
Only support create mv on mv which has SinglePartitionInfo, i will create new pr to support create mv on ExpressionPartitionInfo mv

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|1|1|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/catalog/MaterializedViewTest.java [hunk 1]
```diff
@@ -627,4 +627,48 @@ public void testRollupMaterializedViewWithScalarFunction() throws Exception {
...
+        SinglePartitionInfo singlePartitionInfo = new SinglePartitionInfo();
+        MaterializedView.MvRefreshScheme refreshScheme = new MaterializedView.MvRefreshScheme();
+        HashDistributionInfo hashDistributionInfo = new HashDistributionInfo(3, Lists.newArrayList(columns.get(0)));
+        MaterializedView mv = new MaterializedView(1000, testDb.getId(), "mv", columns, KeysType.AGG_KEYS,
+                singlePartitionInfo, hashDistributionInfo, refreshScheme);
+        List<MaterializedView.BaseTableInfo> baseTableInfos = Lists.newArrayList();
+        MaterializedView.BaseTableInfo baseTableInfo = new MaterializedView.BaseTableInfo(testDb.getId(), baseMv.getId());
+        baseTableInfos.add(baseTableInfo);
+        mv.setBaseTableInfos(baseTableInfos);
+        mv.onCreate();
```
Comment (ABingHuang) [inline fe/fe-core/src/test/java/com/starrocks/catalog/MaterializedViewTest.java:670]: add a create materialized sql to test the nested mv creation.

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java [hunk 2]
```diff
@@ -493,7 +493,12 @@ public void onCreate() {
                 active = false;
                 continue;
             }
-
+            if (table instanceof MaterializedView && !((MaterializedView) table).active) {
+                LOG.warn("tableName :{} is invalid. set materialized view:{} to invalid",
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java:497]: if some base tables are mv, and the mvs is not active, I think it is better to make the creation of mv fail with notice message: "there are some base materialized views are not active" or something?

**Review by ABingHuang — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java (hunk 2 above)
Comment (blackstar-baba) [reply fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java:497]: If do that, will it cause checkpoint and fe restart failed?

**Comment by blackstar-baba** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|4|4|100.00%|[]|

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**